### PR TITLE
fix: correct off-by-one in vocab boundary check for token validation

### PR DESCRIPTION
**Pull request by Ricardo-M-L**
## Summary

- Fix off-by-one error in `_check_vocab_boundary_finish` in `schedule_batch.py`: change `token_id > self.vocab_size` to `token_id >= self.vocab_size`
- Valid token IDs are in range `[0, vocab_size)`, so `token_id == vocab_size` is out of bounds and should trigger the finish condition
- This is consistent with the same boundary check in `spec_utils.py` (line 597) which correctly uses `>=`

## Details

The `_check_vocab_boundary_finish` method is responsible for detecting out-of-bounds token IDs (e.g., caused by NaN in logits or sampling errors). The previous check `token_id > self.vocab_size` would miss the case where `token_id == vocab_size`, which is still an invalid token ID since valid IDs range from `0` to `vocab_size - 1`.

This matches the existing correct pattern in `speculative/spec_utils.py`:
```python
if vocab_size and curr_token_id >= vocab_size:
    accepted = False
```

## Test plan

- [x] Verified consistency with `spec_utils.py` boundary check
- [x] Confirmed `vocab_size` represents the total number of tokens (valid range is `[0, vocab_size)`)
- [x] Confirmed by `clamp_(0, self.model_config.vocab_size - 1)` usage elsewhere in the same file

---
### CI States

Latest PR Test (Base): :hourglass_flowing_sand: [Run #27239246115](https://github.com/sgl-project/sglang/actions/runs/27239246115)
Latest PR Test (Extra): :x: [Run #27239245876](https://github.com/sgl-project/sglang/actions/runs/27239245876)

### CODE REVIEW

**Comment by fzyzcjy** [comment]
Hi, could you please add some tests to ensure no regression, e.g. mimicking (copy-pasting and maybe modifying or maybe implement by yourself freely) https://github.com/sgl-project/sglang/pull/27556
